### PR TITLE
Create build-the-builder.md

### PR DESCRIPTION
**Pull request by cfilleke**
**What this PR does / why we need it**:

Description:
A developer may have need to re-build the bazel builder container image

What you expected:
There needs to be a document describing how to build the builder container for different architectures, discussing how to prepare the builder container build toolchain, how to modify the build scripts for different registry providers, how to update the dependencies in the builder, and how to publish it to a registry for further development use.

**Which issue(s) this PR fixes** 
Fixes #[2049](https://github.com/kubevirt/kubevirt/issues/10494)

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: N/A this is documentation of an heretofore undocumented set of make targets, environment variables and how to update rpm dependencies in building the builder helper container
- [x] Refactor: N/A this is documentation of an heretofore undocumented set of make targets, environment variables and how to update rpm dependencies in building the builder helper container
- [x] Upgrade: Impact of this change on upgrade flows: this PR makes it clearer how to upgrade the rpm dependencies and bazel version in the KubeVirt builder helper container
- [x] Testing: N/A this is documentation of an heretofore undocumented set of make targets, environment variables and how to update rpm dependencies in building the builder helper container
- [x] Documentation: this PR provides documentation of the make targets `make builder-build`, `make builder-publish`, `make bazel-push-images` and `make manifests`
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was made, see: https://groups.google.com/g/kubevirt-dev/c/JqpH4v-gaRE

**Release note**:

```release-note

 - documents steps to build the KubeVirt builder container


```

### CODE REVIEW

File: docs/build-the-builder.md
Comment (EdDev) [inline docs/build-the-builder.md:84]: I see that `KUBEVIRT_BUILDER_IMAGE` is also accepted if an env var is provided.
So I think you do not need to edit the file.

File: docs/build-the-builder.md
Comment (EdDev) [inline docs/build-the-builder.md:96]: I am unclear why there is a need to edit, I think one just needs to specify explicitly the ones that should be updated.

Well, unless the idea is to really update the versions and then push the changes as a PR to update the builder.
But this part is not clear here.

**Review by EdDev — COMMENTED** [review]
Very nice, thank you for this contribution.

Question: Have you or someone else tested these steps and found them working? It will be very nice to specify in the PR description on which environments it was tested.

File: docs/build-the-builder.md
Comment (cfilleke) [reply docs/build-the-builder.md]: Yes. Thank you! :)

File: docs/build-the-builder.md
Comment (cfilleke) [reply docs/build-the-builder.md]: Edited as suggested.

File: docs/build-the-builder.md
Comment (cfilleke) [reply docs/build-the-builder.md]: This procedure was tested on an s390x LPAR running Ubuntu, using the IBM cloud container registry.

**Review by EdDev — COMMENTED** [review]
Thanks, looks great.

To keep the history of the commits clean, could you please squash all your commits and re-push this?
You could wait with this for a while, at least until a maintainer approves the PR.

File: docs/build-the-builder.md
Comment (dhiller) [inline docs/build-the-builder.md:52]: I'd add a note that you can also use [bazelisk](https://github.com/bazelbuild/bazelisk#bazelisk), which automatically downloads and installs the bazel release you need.

See the [.bazelversion](https://github.com/kubevirt/kubevirt/tree/main/.bazelversion) file.

File: docs/build-the-builder.md
Comment (dhiller) [inline docs/build-the-builder.md:124]: How about making this a proper script, as in `./hack/fix-rpm-deps-s490x.sh` and referencing it here? Also I'd suggest using `$1` as input for `arch` then.

File: docs/build-the-builder.md
Comment (dhiller) [inline docs/build-the-builder.md:169]: nit

```suggestion
## Build & push KubeVirt images to registry and make manifests
```

**Review by dhiller — CHANGES_REQUESTED** [review]
Hey @cfilleke ! Thank you for that doc, much appreciated! I have some minor suggestions, nothing that would block the merge of this.

/approve

I suppose @brianmcarey will take a look soon also.

File: docs/build-the-builder.md
Comment (brianmcarey) [inline docs/build-the-builder.md:169]: Everything in this doc seems to be specific to building and publishing the builder image up until this section - I am not sure if this section fits in with the `Build the Builder Container` title?

**Review by brianmcarey — COMMENTED** [review]
/lgtm 

Thanks for this!

Just one question inline.

**Comment by brianmcarey** [comment]
/hold 

Just to give some time for the comments

**Comment by brianmcarey** [comment]
/hold cancel

reread @dhiller's comments and no hold needed here.

**Comment by EdDev** [comment]
I'll prefer the commits to be squashed first.
And I would wait also for feedback to Daniel comments, even if no action will be taken.

/hold

**Comment by cfilleke** [comment]
How does adding a single markdown document cause automated tests of code to fail? 🤔

**Comment by brianmcarey** [comment]
> How does adding a single markdown document cause automated tests of code to fail? 🤔

We have a couple of flaky tests that are causing test lanes to fail. These are being worked on so should improve soon - regarding this PR you shouldn't worry about any failed lanes.

File: docs/build-the-builder.md
Comment (cfilleke) [reply docs/build-the-builder.md:52]: great.  It will be nice when essential tools such as bazel, bazelisk and bazeldnf have regular binary releases for all the architectures we need to use them for.

File: docs/build-the-builder.md
Comment (cfilleke) [reply docs/build-the-builder.md:124]: I'd like to put that in a separate PR, since it solves a different problem, and I think  pinning the dependency versions in the repo will require reviewing the dependencies for security issues.  This is just a suggestion for getting a test version of the builder and KubeVirt operator going for internal test.

File: docs/build-the-builder.md
Comment (cfilleke) [reply docs/build-the-builder.md]: the  environment variables for the successful build of KubeVirt for architectures other than x86 and arm are not in 'getting-started.md' where these make targets are otherwise documented -- should I add them there? Thanks for your advice.

File: docs/build-the-builder.md
Comment (dhiller) [reply docs/build-the-builder.md:52]: Understood that this will not work in arm case. Sorry for my ignorance.

File: docs/build-the-builder.md
Comment (dhiller) [reply docs/build-the-builder.md:124]: Agreed, this can be solved in another PR.

**Comment by dhiller** [comment]
> I'll prefer the commits to be squashed first. And I would wait also for feedback to Daniel comments, even if no action will be taken.
> 
> /hold

Agree. And regarding the release note: the link to the issue is redundant, so I'd remove it and just leave the first line.

File: docs/build-the-builder.md
Comment (cfilleke) [reply docs/build-the-builder.md:52]: so, when bazelisk can automatically download and install the bazel releases for s390x architecture, we can modify this build-the-builder.md

Comment (cfilleke) [reply docs/build-the-builder.md:52]: trying to figure out how to 'resolve conversation' here

**Review by dhiller — CHANGES_REQUESTED** [review]
Hey @cfilleke, thanks for squashing everything into one commit. 
However, now the commit message doesn't describe what it changes. Would you be so kind and change it to a brief description of what you did?

**Comment by cfilleke** [comment]
> Hey @cfilleke, thanks for squashing everything into one commit. However, now the commit message doesn't describe what it changes. Would you be so kind and change it to a brief description of what you did?

Thanks, amended the commit message.

**Comment by cfilleke** [comment]
Thanks, Edward!

This procedure was tested on an s390x LPAR running Ubuntu, using the IBM
cloud container registry.

**Review by dhiller — APPROVED** [review]
Great work, thank you @cfilleke !

/approve

**Comment by dhiller** [comment]
@cfilleke apologies, I didn't see the `hold` label.

**Pull request merged**
